### PR TITLE
Update target.md

### LINKED ISSUE #1112: Update/improve docs on `target`
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/dbt-jinja-functions/target

https://docs.getdbt.com/dbt-cli/configure-your-profile#about-profiles

### What part(s) of the page would you like to see updated?

I had a customer reach out indicating that the target documentation (first link) "implies that for every case the `target` can be defined in Profile, which is not true for our adapter (Spark)." This makes me think there is particularly confusion around profiles and how they work in dbt Cloud.

I think it would be good to include a link on the main target page (first link in the above box) to the profiles page (second link). 

Additionally, I think it would be really helpful to provide more context/examples of how target/profiles translate in dbt Cloud. I find that theses concepts are not super easy to explain to customers and is better understood (for me, at least) by playing around with targets to see how things change as targets change. I feel like showing examples of the impact of different targets in dbt Cloud specifically could help clear some of this up for customers.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by runleonarun**
Closes https://github.com/dbt-labs/docs.getdbt.com/issues/1112

## What are you changing in this pull request and why?
* Adding more information about Cloud setup for target names, making it clear that there are different approaches for different adapters.
* Also updating the content so it's current.
* updated a few links to be more direct and relevant
*

### CODE REVIEW

File: website/docs/reference/dbt-jinja-functions/target.md [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
-* **dbt Cloud Scheduler:**
-    * `target.name` is defined per job as described [here](/docs/build/custom-target-names).
-    * For all other attributes, the values are defined by the deployment connection. To check these values, click **Deploy** from the upper left and select **Environments**. Then, select the relevant deployment environment, and click **Settings**.
-* **dbt Cloud IDE:** The values are defined by your connection and credentials. To check any of these values, head to your account (via your profile image in the top right hand corner), and select the project under "Credentials".
+- **dbt Core:** These values are based on the target defined in your [profiles.yml](/docs/core/connect-data-platform/profiles.yml) file. Please note that for certain adapters, additional configuration steps may be required. Refer to the [set up page](/docs/core/connect-data-platform/about-core-connections) for your data platform.
+- **dbt Cloud** To learn more about setting up your adapter in dbt Cloud, refer to the [About data platform connections](/docs/cloud/connect-data-platform/about-connections).
```
Comment (runleonarun) [inline website/docs/reference/dbt-jinja-functions/target.md]:
```suggestion
- **dbt Cloud** To learn more about setting up your adapter in dbt Cloud, refer to [About data platform connections](/docs/cloud/connect-data-platform/about-connections).
```

File: website/docs/reference/dbt-jinja-functions/target.md [hunk 2]
```diff
@@ -1,20 +1,18 @@
 ---
-title: "About target variable"
+title: "About target variables"
 sidebar_label: "target"
 id: "target"
-description: "Contains information about your connection to the warehouse."
+description: "The `target` variable Contains information about your connection to the warehouse."
```
Comment (mirnawong1) [inline website/docs/reference/dbt-jinja-functions/target.md]:
```suggestion
description: "The `target` variable contains information about your connection to the warehouse."
```

File: website/docs/reference/dbt-jinja-functions/target.md [hunk 3]
```diff
@@ -1,20 +1,18 @@
...
-* **dbt Cloud IDE:** The values are defined by your connection and credentials. To check any of these values, head to your account (via your profile image in the top right hand corner), and select the project under "Credentials".
+- **dbt Core:** These values are based on the target defined in your [profiles.yml](/docs/core/connect-data-platform/profiles.yml) file. Please note that for certain adapters, additional configuration steps may be required. Refer to the [set up page](/docs/core/connect-data-platform/about-core-connections) for your data platform.
+- **dbt Cloud** To learn more about setting up your adapter in dbt Cloud, refer to [About data platform connections](/docs/cloud/connect-data-platform/about-connections).
+   - **dbt Cloud Scheduler:** `target.name` is defined per job as described in [Custom target names](/docs/build/custom-target-names). For other attributes, values are defined by the deployment connection. To check these values, click **Deploy** and select **Environments**. Then, select the relevant deployment environment, and click **Settings**.
```
Comment (mirnawong1) [inline website/docs/reference/dbt-jinja-functions/target.md]: might be good to link to the scheduler
```suggestion
   - **[dbt Cloud Scheduler](/docs/deploy/job-scheduler)**: `target.name` is defined per job as described in [Custom target names](/docs/build/custom-target-names). For other attributes, values are defined by the deployment connection. To check these values, click **Deploy** and select **Environments**. Then, select the relevant deployment environment, and click **Settings**.
```

File: website/docs/reference/dbt-jinja-functions/target.md [hunk 4]
```diff
@@ -1,20 +1,18 @@
...
+   - **dbt Cloud IDE:** These values are defined by your connection and credentials. To edit these values, click the cog in the top right, select **Profile settings**, and click **Credentials**. Select and edit a project to set up the credentials and target name.
```
Comment (mirnawong1) [inline website/docs/reference/dbt-jinja-functions/target.md]: adding link to ide docs and also i used to use 'cog' and was been given feedback that we use the term 'gear icon' instead.


```suggestion
   - **[dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud)**: These values are defined by your connection and credentials. To edit these values, click the gear icon in the top right, select **Profile settings**, and click **Credentials**. Select and edit a project to set up the credentials and target name.
```

Comment (mirnawong1) [reply website/docs/reference/dbt-jinja-functions/target.md]: also might be helpful to have screenshots for scheduler and ide but not a blocker or urgent

**Review by mirnawong1 — APPROVED** [review]
small feedback and approved for you @runleonarun !

**Comment by runleonarun** [comment]
Thanks for the review, @mirnawong1!

**Pull request merged**